Offline bundle

pjanik opened the pull request:
@scytacki, it's based on your idea. I didn't modify JSmol.full.js but monkey patched `Jmol._getFileData` instead. Hopefully readme explains other bits.

I also deploy this package to S3, so it can be used by http://lab.concord.org/standalone/create.html page.

pjanik committed:
Upgrade iframe-phone to v1.2.0

[#139951055]

pjanik committed:
Implement a script that bundles JSmol into a package that works offline

[#139951055]

pjanik committed:
Add offline bundle config (so small ver can be created), build and deploy offline version to S3

[#139951055]

pjanik committed:
Update / fix TravisCI config and handle relative paths better

[#139951055]

pjanik committed:
Add offline-bundle readme

[#139951055]

pjanik committed:
Add alert message if some file is missing

[#139951055]

scytacki commented on embeddable.html:
```diff
@@ -38,6 +38,12 @@
 <script type="text/javascript" src="iframe-phone/dist/iframe-phone.js"></script>
 <script type="text/javascript" src="shutterbug.js/dist/shutterbug.js"></script>
+<!--
+  Do not remove comment before. It's used by offline-bundle app.
```
Should 'before' be changed to 'below'?

scytacki reviewed (approved):
Looks great to me.

scytacki commented:
One thing I didn't fully understand was the binary file handling. They are just copied, right?  
So how are images handled? Does JSMol create `img` tags or use canvas elements, and those things work with file URLs?  
I thought non image binary files were loaded by JsMol using its proxy service, and that would fail when offline. Or do I have this all wrong?

pjanik committed:
Fix comment

[#139951055]

pjanik commented:
> One thing I didn't fully understand was the binary file handling. They are just copied, right?
So how are images handled? Does JSMol create img tags or use canvas elements, and those things work with file URLs?

So, binary, non-image files are not handled. I just skip them for now. We don't use any at all, so I don't have anything to play with. Hopefully we could handle them if we convert them to base64 (JSmol logic is more complex, but that's what I'd hope for if we ever needed that).

Images are handled separately. My guess is that JSmol loads them using Image() class, not using _getFileData - `file://` protocol is supported anyway. So, I just copy all the images from `/models` dir and keep the original paths. It should work, I've tested that.

Actually, I might change approach later. Instead of bundling everything into HMTL, I might try to generate JS files with something like `Jmol._fileData['models/dna/dna.pdb'] = '<file_content>'`. It might make it easier to add new, custom model files. We might need it for PASCO JSmol template interactive. Currently, they'd have to edit 20MB HTML file - it take a while to open it in editor and it's not very readable. The only problem is that it'd be nice to use new js strings (``) to handle multiline strings easily, but we'd support less browsers (probably IE can't handle it).

pjanik committed:
Fix _getFileData.offline-patch, update offline bundle config

[#139951055]

- Additional \n was breaking jsmol in some cases
- Add more files to offline bundles (they seem necessary)

pjanik commented:
I've fixed some small bugs, so I'm going to merge it now.
